### DISCUSSION TITLE
Zc95 freeze when wifi enabled

### DISCUSSION THREAD

**Discussion by Flyer133** [CrashOverride85/zc95#25] [Q&A] [answered]
Hello!
I apologize, my english is not good because i'm french...
I have done the zc95 box, the build instructions was genius!  Really easy sold and assemble the box.
All  functions are Ok, but impossible to use wifi.
My zc95 use 2 pico-w.
When i try to configure wifi using "Config Wifi/AP mode" the board freeze. Impossible to do anything, reboot.
First, i tried to invert the pico boards. The same problem.
All is ok, but not wifi and i dont know why...
Have you an Idea?
Regards
Julien

**Comment by CrashOverride85** [comment]
Hi, thanks for the message, glad to hear you've built one and it's mostly working!

I've been playing about with the "Config Wifi/AP mode" menu, and the only way I can reproduce this is if you start AP mode, exit, then try again. That part is a bug and I'll have a go a fixing it, but I'm unless you are going into "Config Wifi/AP mode", exiting, then going back in again I doubt that's the issue :(

BTW, which screen is it freezing on, is the "Starting..." one?

It does look like there's a few improvements I can make to the AP mode setup, though, especially when it comes to scanning for WiFi networks, and showing more info (e.g. number of networks found) on the "Starting..." screen.

**Comment by Flyer133** [comment]
Hello!
Thanks you for your quick answer!
Next you took the link to show the vidéo of the error...

https://youtube.com/shorts/QQ2FELe7N34?feature=share

If you have an Idea...
When the box is powered, the led D7 (OK_LED) is off.
Regards!

**Comment by CrashOverride85** [comment]
Thanks for the video, that makes it clearer exactly where it's going wrong.

I've been testing my box, and I can't get to fail in that way, so I'm a bit confused about why it's going wrong.
Would you be able to get the serial debug output from the box when it hangs?

There's two ways to do this:
1. If you've got a 232 level USB to serial adaptor, serial out is on pin 3 with ground on pin 5 of the DB9 accessory port.

2. With a 3v3/ttl serial adaper, it's a bit harder unfortunately. Serial TX can be accessed on J8 (the one with the jumpers), pin 7 (labelled ">TTL-TX"), and use a ground from the opposite header (J17).
To get debug on this pin, you also need to go to Config >  Hardware Config > Debug output and change it to "Aux port" (default is "Accessory port" which the first option needs)

For both options, baud rate is 115200.


About the OK_LED - for the time being, that's normal. The LED isn't used at the moment, but thanks for reminding me, I really do need to update the code to use that LED.
BTW if the output board wasn't ok, you would see an error on screen the moment you powered it on.

**Comment by Flyer133** [comment]
Hey CO85!

Here is the serial debug output:

~CMenu()
Loop time: 516, batt: 100
Loop time: 526, batt: 100
CMenuRemoteAccess() 
Loop time: 522, batt: 100
Loop time: 521, batt: 100
Loop time: 514, batt: 100
Loop time: 514, batt: 100
CMenuApMode()
CAnalogueCapture::stop()
CAnalogueCapture::stop() done
CMenuApMode::wifi_scan(): Start scanning for wifi networks
CWifi::init(): wifi/cyw43 init success

and at this precise moment a low but very high frequency sound is emitted by the circuit and all is freezing!

Thanks for your answers, if you have an idea...
Julien

**Reply by Flyer133** [reply]
else...
does it exist an other way to upload LUA code?

**Reply by CrashOverride85** [reply]
The very next thing it tries to do after that serial output is wifi scanning, so the only thing I've been able to think of that /might/ cause it is a problem with the 5v supply.
AFAIK when WiFi is enabled, power consumption of the Pico W briefly jumps significantly, so if there's a problem with the supply, it might cause it to lock up.

This could probably be ruled out by connecting the Pico to USB for power as well, if it then works, then it's likely a power issue with the 5v rail. 
You would need to power up the zc95 as normal (from battery) first, then connect the pico on the main board to USB power, then try starting ap mode / wifi config (external USB power will stop the pico resetting, so power cycling the box with USB connected will produce weird results).

Right now WiFi is the only way to get Lua scripts on to it, but I'll see if I can update it to allow serial upload

**Reply by Flyer133** [reply]
It works! It was the solution! 
Thanks CrashOverride85!
Julien

**Reply by CrashOverride85** [reply]
Cool, glad it's working :)
But it sounds like something's up with the 5v supply.
Although it might be worth trying adding something like a 100nF and 100uF capacitor in parallel across pins 39 (VSYS) and 38 (GND) of the Pico, to see if that lets it work without USB power

**Comment by jlatex** [comment]
I had the exact same issue.  I don't have a battery yet, just using a 15v supply.   Powering the pico via usb allowed it to connect.  Adding 100uF/100nF across the pico didn't help

**Comment by jlatex** [comment, marked as answer]
Okay, so I connected the VSYS line direct to 5v (bypassing the bs250) and then it works.  I noticed the data sheet for the pico suggests a mosfet with Ron of 100mR and the bs250 is 8R so perhaps we need a better mosfet here.

**Comment by CrashOverride85** [comment]
Good spot, thanks 🙂

I think the easy answer is going to be to to use a schottky diode between VSYS and the 5v supply, in place of the MOSFET. The 5v supply is already quite a bit more than the minimum 1.8v required by the Pico, so the diode drop shouldn't be a concern, especially as the Pico is only powering itself.
